bitcoin/src/lib.rs

```diff
@@ -13,21 +13,28 @@
 //! See README.md for detailed documentation about development and supported
 //! environments.
 //!
-//! ## Available feature flags
+//! # Available feature flags
 //!
-//! * `std` - the usual dependency on `std` (default).
+//! * `default` - enables `std` and `secp-recovery`.
+//! * `std` - the usual dependency on `std`.
+//! * `serde` - implements `serde`-based serialization and deserialization.
+//! * `secp-lowmemory` - optimizations for low-memory devices.
 //! * `secp-recovery` - enables calculating public key from a signature and message.
+//! * `bitcoinconsensus` - (dependency) enables validation of scripts/transactions by `rust-bitcoinconsensus`.
+//! * `bitcoinconsensus-std` - same as `bitcoinconsensus` but also enables `std`.
+//! * `rand` - makes it more convenient to generate random values.
+//! * `rand-std` - same as `rand` but also enables `std`.
 //! * `base64` - (dependency), enables encoding of PSBTs and message signatures.
-//! * `rand` - (dependency), makes it more convenient to generate random values.
-//! * `serde` - (dependency), implements `serde`-based serialization and
-//!                 deserialization.
-//! * `secp-lowmemory` - optimizations for low-memory devices.
-//! * `bitcoinconsensus-std` - enables `std` in `bitcoinconsensus` and communicates it
-//!                            to this crate so it knows how to implement
-//!                            `std::error::Error`. At this time there's a hack to
-//!                            achieve the same without this feature but it could
-//!                            happen the implementations diverge one day.
 //! * `ordered` - (dependency), adds implementations of `ArbitraryOrdOrd` to some structs.
+//!
+//! # Note on `foo-std` features
+//!
+//! Feature pairs of form `foo`/`foo-std` are additive (i.e. can both be enabled in a build) however
+//! it is expected that users pick one, for std builds use the `foo-std` feature and only use the
+//! plain `foo` feature for no-std builds. All `foo-std` features also enable `std` so you don't
+//! need both. At times there is some trickery involved to keep the features additive so using the
+//! no-std feature in a std build may result in missing functionality e.g., error message content
```

Comment:
```suggestion
//! `no-std` feature in a std build may result in missing functionality, e.g. error message content
```

Reply:
Ha! I knew you would mention backticks on the no-std, you missed the std :) I should have justified it really though, I intentionally left them off std/no-std because we do not have a no-std feature so I thought the pair were words not code? On a related point, I usually use "feature" instead of `feature` but I'm not sure if that is good practice or worth differentiating?

Re the eg thing, I spent some time a few years back trying to learn how to use eg, it was one of the first times I really geeked out on grammar (after a couple years or incorrectly using double spaces after full stops). I came to the conclusion that there are multiple accepted "correct" ways and  _I think_ it depends on the type of publication. I elected for the current form and then in parenthesis no coma 

- I use comas when there are no parenthesis e.g., like this.
- I omit the coma when using parenthesis (e.g. like this).

Both of which are quite formal, I think our docs should be formal. In normal rushed writing I just write it shorter eg, like this.

I never saw the coma first like you suggest, e.g. like this?

As usual, happy to be schooled, I don't mind how we do it as long as there is uniformity and I can follow a rule so my OCD doesn't trigger.

Reply:
Sorry about the wall of text :)

Reply:
Oh I never went into a rabbit hole with "e.g." but feel free to opt for any solution you find useful.

Regarding the backticks, here's my rationale when to use it:

- everything that is code related like a function or a type. Example `Address::from_str(...)`
- everything that you have to type in a terminal. Example `cargo add bitcoin -F rand-std`

Now why did I backticked just the "no-std" and not the "std" in the sentence:

> At times there is some trickery involved to keep the features additive so using the `no-std` feature in a std build may result in missing functionality e.g., error message content

Because when reading this I feel that the "no-std feature" is a feature in a sense that I can do `cargo add bitcoin -F no-std`, hence it deserves a backtick because it is something I type in the terminal.
In the case of the "std build" it is not implying that I "type it in a terminal" explicitly like `cargo add -F std` because it is more generic.
It is a "build", so it could be `cargo add -F rand-std` or `cargo add -F bitcoinconsensus-std`.

Now I realize that I am wrong because we don't have a feature flag `no-std`.
We do have "no-std builds" with `--no-default-features`, due to the fact that `std` in

https://github.com/rust-bitcoin/rust-bitcoin/blob/a132cef572c1796ea469f0a6b537d7e46ac5383d/bitcoin/Cargo.toml#L18

won't be "activated".

Did you get my point?
I will try to summarize it:

- Don't use "no-std feature" since we don't have it. Use "no-std build".
- Backtick stuff that might be typed in a terminal with `cargo add -F foo`, i.e. backtick all formal feature flags.

You are forgiven for your wall of text, if you forgive my wall of text.

Reply:
I see your wall of text and raise you one :)

Yep, I get you. Nice one.